### DISCUSSION TITLE
FreeDCam as a professional camcorder app.

### DISCUSSION THREAD

**Discussion by massimostella** [KillerInk/FreeDcam#162] [Ideas]
Hi all,
After some tests on some smartphones (mostly cheap ones around $200 - my main one is now a Mi10T Lite) I can tell that FreeDCam was always able to use all the available lenses and sensors at all possible resolutions and framerates.
This is not possible with Open Camera and HedgeCam2 which are able to use only the main camera and, even if they allow you to set the framerate parameter to 24 or 25 fps this is always ignored during the recording. This issue doesn't occur in these 2 applications for time-lapse and slow motion recording.
The only applications which are able to actually modify the framerate are FilmicPro (that on Android is only able to uses the main camera) and MC24Pro which is the only application around that is able to do what FreeDCam is able to do.
This is what shown me that FreeDCAm is actually a possible replacement for these professional tools for filmmakers except for few limits I've found:
1.  Actually this is one of the things that has shown me that FreeDCam is mostly thought for still photography than for video: The Focus and the Zoom control, even if I admit that they are very precise, don't allow you to pass from the 2 opposite positions (e.g. macro to infinite or no Zoom to maximum zoom) in a single finger movement because you have to scroll more than one time to walk all the lenght. This is OK if you set the camera before to shoot, but it's not good if you have to create a smooth shift during a video recording.
2. A similar issue, due to the "photo camera approach", occurs with the magnification during focusing: this is very helpful, and you can find it in professional camcorders, but the magnification is on the preview only and not in the final recording. 
3. FreeDCam lacks of S-Log, C-Log, Gamma and other curves as recording profiles. It's possible to find some of these curves already implemented in Open Camera and in HedgeCam2 (maybe it's possible to bring the code from these applications). They are very well implemented in MC24Pro and if you want to have them in FilmcPro you have to pay an extra. These profiles are used to extend the exposition range to have all the 12 stops inside the 5 stops of the REC709. This for video makers is what it is RAW pictures for a photographer.
4. The white balance system is not bad but presets are a little too rigid for video makers who are used to have the manual control via Kelvin degrees control or RGB or by making a reading of the light in the environment. In Open Camera the manual control with a slider to change the temperature of the Kelvin degrees work pretty well and in HdegCam2 the RGB control is already implemented and it works great (maybe it's possible to take inspiration from this code too)
5. In FreeDCam, except for the source selection, there are not any audio controls or VU-meters to monitor it. A volume control and a way to monitor it graphically but on the headphones too (which you can activate/deactivate) would be really useful.

Once the items I've listed will be implemented FreeDCam will be a perfect tool for filmmakers. The only feature it'll miss that FilmicPro and MC24Pro have are the ability to set automations like focusing, zoom, iso and shutter speed. (Automated focus pulling is very useful)
In these applications you can set the starting parameter, the ending parameter and the speed you want to use to pass form one parameter to the other one. When you need that the operation begins you have to hit the automation button only and voila!

To complete the list with all possible options/feature, a Gimbal interface could really be a plus but this is a very rare and available in Filmic Pro only, for the moment.

I'm waiting for feedback, and I'm available to go in deep in any of the topic I've listed if needed and if someone is interested in.

Cheers.

Massimo.

**Comment by KillerInk** [comment]
> Actually this is one of the things that has shown me that FreeDCam is mostly thought for still photography than for video: The Focus and the Zoom control, even if I admit that they are very precise, don't allow you to pass from the 2 opposite positions (e.g. macro to infinite or no Zoom to maximum zoom) in a single finger movement because you have to scroll more than one time to walk all the lenght. This is OK if you set the camera before to shoot, but it's not good if you have to create a smooth shift during a video recording.

That can get added.

> A similar issue, due to the "photo camera approach", occurs with the magnification during focusing: this is very helpful, and you can find it in professional camcorders, but the magnification is on the preview only and not in the final recording.

Not sure if that makes sense. Its then burned into the recorded movie.
Funny thing is with using renderscript as preview postprocessing, on some phones it happen that it get recorded.

> FreeDCam lacks of S-Log, C-Log, Gamma and other curves as recording profiles. It's possible to find some of these curves already implemented in Open Camera and in HedgeCam2 (maybe it's possible to bring the code from these applications). They are very well implemented in MC24Pro and if you want to have them in FilmcPro you have to pay an extra. These profiles are used to extend the exposition range to have all the 12 stops inside the 5 stops of the REC709. This for video makers is what it is RAW pictures for a photographer.

you can create your own curves for freedcam. 
one for all channels and also seperate each channel. that file must get named tonecurveprofiles.xml 
and placed inside your **sdcard/Android/data/troop.com.freedcam/files/**
content of the file looks like this
```
<tonecurves>
    <tonecurve name= "linear">
        <rgb>0.0,0.0,1.0,1.0,</rgb>
        <r>0.0,0.0,1.0,1.0,</r>
        <g>0.0,0.0,1.0,1.0,</g>
        <b>0.0,0.0,1.0,1.0,</b>
    </tonecurve>
</tonecurves>
```
after that you can enable it inside settings with changing then tonemap mode to contrast curve
then a new icon appears inside manuals wich finaly let you choose the curve

> The white balance system is not bad but presets are a little too rigid for video makers who are used to have the manual control via Kelvin degrees control or RGB or by making a reading of the light in the environment. In Open Camera the manual control with a slider to change the temperature of the Kelvin degrees work pretty well and in HdegCam2 the RGB control is already implemented and it works great (maybe it's possible to take inspiration from this code too)

FreeDcam have it too. In the cameraui click on the temperature icon(left side top) and turn it off. after that it appears in the manuals.


> In FreeDCam, except for the source selection, there are not any audio controls or VU-meters to monitor it. A volume control and a way to monitor it graphically but on the headphones too (which you can activate/deactivate) would be really useful.

Not sure if the audio api is so powerfull. Never went into that direction

About automation:
think that should be possible todo.

greets Ingo

**Reply by massimostella** [reply]
Hi Ingo,
I had seen already the contrast curve in the tone map settings and I tried to manage it but in my configuration there is no way to show up the icon you wrote about and to select a curve. If I'll be able to use this tool I'll try to write all the useful curves for video production.

After I read your detailed explanation I checked all the listed icons and I saw that some of them are not shown even if I set the camera to use that functions. Here is the list of these "missing" icons/functions:
- Manual Aperture/F~Number (but in a phone camera you generally can't set it because the aperture is fixed)
- Manual Brightness
- Night Mode
- Continuous Capture Mode
- HDR Mode
- Contrast
- Program Shift

About the temperature, you are right. I just forgot about this because as soon as I turn it on the colors turn to green. But this time I saw that if I scroll to change the temperature everything works as expected.
The issue is that when in manual, all my settings are reset as soon as I stop a recording or if I pause the application. Reset means that the ISO goes to Auto and keeps 50 as a value, the shutter speed is moved to 1/10000 and the white balance to 1500, but it actually renders a green picture up when I don't touch the scroll bar to change the value. 
The only way to avoid this is to keep FreeDCam on Auto and lock/unlock the AE priority.

I know that this actually depends on my smartphone because of its API, but I'm glad you are using less stable but more open and experimental code than other projects I've quoted.
I remind you that I'm testing the camera on a Mi10T Lite (https://www.mi.com/global/mi-10t-lite/specs/) and I'm able to record videos at a bunch of different sizes and framerates using customized profiles (if you want the full list lest me know)
If you want more information or there is a way to dump some useful datas from the phone let me know. I'd like to avoid rooting it if possible.

About the focus and zoom and other control which are in the scroll bar and possible automations I suggest considering take inspiration from the approach which FilmicPro and MCPro24fps apps are using by adopting wheels.
Here you can watch in 2 minutes how they work: https://www.youtube.com/watch?v=AlLapwSQ7uw

For which concern the audio a VU meter to help monitor the audio (if you knock on a microphone, and you see that the meter is bumping you can understand which source is active and at which volume or if your sources are all inactive, for e.g.) could be a great help already. 

To complete what I wrote last time for having a list of all the few missing features for video (but these could be useful for still photo too) it's important to not forget preview modalities for Anamorphic and DOF adapters: the first one asks for a way to squeeze down the image to the correct aspect ratio and the second one to reverse the image upside down.

Cheers,

Massimo.

**Comment by massimostella** [comment]
Short update:
No Scene or Color Effect is working not even when I switch the Camera Control to Scene.

Cheers.

Massimo

**Reply by KillerInk** [reply]
scene and color works here on all my phones. think you have the same behavior on other apps.
only the tonecurve is broken.

**Comment by massimostella** [comment]
Color actually works but if I use the RenderScript only. If the Preview PostProcessing is not activated or OpenGL is on, no color effect is applied.
RenderScript in my phone as the following issues:
1. The image from the selfie camera is shown upside down
2. When you switch camera, frequently FreeDCam is not able to start the new device and it enters in a loop bouncing from video mode to picture mode while the screen is black and the icons blink. If you restart the application you can complete the switching operation. Sometime, when I turn on the app, it starts in the loop already. One or two restarts always fix the issue.

For which concern the Scene, I saw that in Open Camera and HedgeCam2 when I switch the Scene a little change in the previewed image is visible. As in FreeDCam the screen is covered by the menu I can't tell you if something happens. My procedure is to set the Camera Control to Scene and I change the Scene. My expectation is to have some icon or something that remind me that I'm in HDR for example, but I have no visual evidence of it and I can't see any difference in the pictures that I take.

Sadly here I've only another phone, a Redmi Note 2 (very old) and the Redmi note 7 I had is now broken. I'm asking to friends to install the application so that they can report me and I can collect more info. I hope this can help.

**Reply by KillerInk** [reply]
color mode is not depending on previewpostprocessing. thats a camera api feature. 
i know that currently not everything is working and i promisse it will get worser^^
currently im changing alot ui stuff. wait for the next release and maybe its working then better

**Reply by massimostella** [reply]
You are doing a great work, and I'm sure things will improve but the number of phones around and the difference in the hardware are so many that I can imagine that it's not easy. Sadly, I can't help you more than testing on my device and by asking other people to do install the app on their phone. What I can tell you is that even if the preview processing is not involved when I'm testing the phone the color mode behave differently in relation to which one is active or not (if you want I can do a screen recording to show you this). I'd like to be more helpful so let me know what I can do to support you better.
Cheers.
